internal/witness/cmd/client/main.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// client fetches and verifies new STHs for a set of logs from a single witness.
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+	ct "github.com/google/certificate-transparency-go"
+	wit_api "github.com/google/certificate-transparency-go/internal/witness/api"
+	wh "github.com/google/certificate-transparency-go/internal/witness/client/http"
+	"github.com/google/certificate-transparency-go/internal/witness/verifier"
+	"github.com/google/certificate-transparency-go/loglist3"
+)
+
+var (
+	logList   = flag.String("log_list_url", "https://www.gstatic.com/ct/log_list/v3/log_list.json", "The location of the log list")
+	witness   = flag.String("witness_url", "", "The endpoint of the witness HTTP API")
+	witnessPK = flag.String("witness_pk", "", "The base64-encoded witness public key")
```

Comment:
Might be worth adding `if *witness == "" { // error` checks for these at the top of `main` (just to be super helpful with accurate error messages)?

Reply:
Done, I've also added a missing check to the feeder.